src/Bicep.Core/Syntax/ModuleDeclarationSyntax.cs

```diff
@@ -41,6 +42,11 @@ public ModuleDeclarationSyntax(IEnumerable<SyntaxBase> leadingNodes, Token keywo
         
         public SyntaxBase Value { get; }
 
+        /// <summary>
+        /// The URI of the JSON template from which the module will be converted. This is only used in the decompiler.
+        /// </summary>
+        public Uri? TemplateUri { get; }
```

Comment:
>TemplateUri [](start = 20, length = 11)

Can we avoid adding this?

Reply:
Yeah, this is definitely not an ideal approach. Added a dictionary to `TemplateConverter` to store json template URIs.